### PR TITLE
Change azure_group_exists with a  minor refactoring

### PR DESCRIPTION
**Pull request by thiagocaiubi**
Fetch group by name instead of listing all resources groups then iterate over them to check if given resource group exists.

### CODE REVIEW

**Comment by thiagocaiubi** [comment]
@katcipis Still, didn't get why it returns "0" when condition is true and "1" otherwise.

**Comment by tiago4orion** [comment]
C programmers ...

**Comment by thiagocaiubi** [comment]
@tiago4orion thanks! 

```console
$ true
$ echo $?
// 0
$ false
$ echo $?
// 1
```

**Comment by tiago4orion** [comment]
nash doesn't have a bool type yet, then we need to emulate it in someway. In C, the POSIX functions commonly succeed with 0 and any other number in case  of fail. The POSIX defines the exit status of a program to be 0 in case of success and any other uint8 number for fail also.

Nash will have boolean types soon, then we'll have better support for this.

File: azure/group.sh [hunk 1]
```diff
@@ -24,11 +24,11 @@ fn azure_group_get_names() {
...
 # already exists, "1" otherwise.
 fn azure_group_exists(name) {
-        groups <= azure_group_get_names()
-        for group in $groups {
-            if $group == $name {
-                return "0"
-            }
-        }
-        return "1"
+	group_name <= az group show --name $name | jq -r ".name"
```
Comment (tiago4orion) [inline azure/group.sh]: Nash aborts if some program fails (return $status != 0) and user didn't check the status.
I don't know if this is the case, but if az program fails if group doesn't exists then nash will abort at this statement also.
You can just rely on $status variable to know if the group exists or ask Nash to ignore the status to not abort.

```sh
fn azure_group_exists(name) {
    # only checking status
    _, status <= az group show --name $name 
    return $status
}
```
or like you did but asking Nash to ignore error:
```sh
fn azure_group_exists(name) {
    groupName, _ <= az group show --name $name | jq -r ".name"
    if $groupName == $name {
        return "0"
    }
    return "1"
}
```

Comment (thiagocaiubi) [inline azure/group.sh]: For this particular case `az group show` exists with `0` regardless if a group exists or not.

`nash` repl:
```console
λ> all <= azure_group_get_names()
λ> echo $all
xpto
λ> xpto_exists <= azure_group_exists("xpto")
λ> echo $xpto_exists
0
λ> ijk_exists <= azure_group_exists("ijk")
λ> echo $ijk_exists
1
```

`az` cli:
```console
$ az group show --name xpto
$ echo $?
// 0
$ az group show --name ijk
$ echo $?
// 0
```

Despite that I agree with you `klb` should be more reliable.

Comment (tiago4orion) [inline azure/group.sh]: wtf... az sucks...

What you think about passing -e to `jq` as we talked earlier? Then we can just return the $status variable.

Thanks for your help!

Comment (lborguetti) [inline azure/group.sh]: @tiago4orion  The idea of using the -e option is because jq fail when resource_group don't exist and return the status to nash?

**Comment by katcipis** [comment]
@thiagocaiubi one of the first proposals that I made to @tiago4orion was to add booleans on nash =), followed by integers =P. I was very torn when I started to code on klb, to be consistent with C style logic, or with posix process error code logic. In the end I chose consistency with the posix logic to avoid code that verified success using **== "0"** and other code that used **!= "0"**.

Neither options actually make me happy =)

File: azure/group.sh (hunk 1 above)
Comment (tiago4orion) [inline azure/group.sh]: Yep

Comment (lborguetti) [inline azure/group.sh]: Hum! Nice :+1:

**Pull request merged**